### DISCUSSION TITLE
Unattended mode

### DISCUSSION THREAD

**Discussion by chronos1970** [farag2/Sophia-Script-for-Windows#282] [General]
I would like to use this script with **MDT** to create Windows images. However, I currently don't see a way to use functions that use prompts in an **unattended** way.

Maybe we can specify answers in a text file or using arguments. I hope that such a feature will be added, as this is the only thing stopping me from using this amazing script.

**Comment by farag2** [comment]
Unfortunately, there is no way, unless I have to re-write the whole code. If you know PowerShell, you may create your own script by taking every needed function. Only this way. The script wasn't designed to use in MDT origionally.

**Comment by itjf-de** [comment]
@chronos1970 i had the same idea to use this amazing script in MDT and today i ran the script with the desired settings and commented out everything that was interactive and relevant for me (all changes were made in the module file "Sophia.psm1" like in line 191 to 218 and 12396 to 12425) 

https://github.com/farag2/Sophia-Script-for-Windows/blob/058e144f48f6685cedfb7985e7cca49112c274d0/Sophia%20Script/Sophia%20Script%20for%20Windows%2010/Module/Sophia.psm1#L191

besides that i disabled "Check if the current module version is the latest one" (line 156 to 186) so that my changes won't be overwritten

of course there is no guarantee of completeness as i am not using all function, respectively others than you

**Reply by farag2** [reply]
Ah, a hacker you are. :)

**Reply by itjf-de** [reply]
not really, i'm just a bit used to powershell xD

however, in a revision of the script it would be awesome to have a build in non-interactive mode

**Reply by farag2** [reply]
You really suggest to add a special key for `.\sophia.ps1 -NonInterractive` to let the script run without any prompts? But how will you manage with UWP apps? There is a GUI, and not only there.

**Reply by itjf-de** [reply]
i must confess, that i have not used the UWP stuff and never seen the popup-dialog before 

of course, that does not make it any easier - however, it would still be possible to make appropriate decisions in a configuration file.
in terms of UWP, it would of course not be possible to see what is actually running on the pc, but if you specify for example that the calculator is to be removed, the script could still try to uninstall it even if it's not installed

but i realize that this is extremely getting out of hand and a ton of work..

**Reply by farag2** [reply]
Much more than you can imagine. May be in the future:)
